pkg/daemon/daemon.go

```diff
@@ -393,6 +433,51 @@ func (dn *Daemon) Run() {
 			if err != nil {
 				glog.Errorf("linuxPTP apply node profile failed: %v", err)
 			}
+		case event, ok := <-saFilesWatcherEventCh:
+			// File system event on sa_file directory detected, restart PTP processes
+			if !ok {
+				glog.Error("fsnotify watcher channel closed, disabling sa_file monitoring")
+				saFilesWatcherEventCh = nil
+				continue
+			}
+			if event.Op&(fsnotify.Write|fsnotify.Create|fsnotify.Remove) == 0 {
+				continue
+			}
+			if strings.HasPrefix(filepath.Base(event.Name), ".") {
+				continue // Ignore hidden files like .data
+			}
+			glog.Infof("Security file changed: %s (op: %s), restarting PTP processes", event.Name, event.Op.String())
+			err := dn.applyNodePTPProfiles()
```

Comment:
Kind of concerned here that applyNodePTPProfiles could theoretically be called twice in quick succession, which could have race conditions that applyNodePTPProfiles isn't hardened for.   Currently we are looking at a ticker and check for updates based off that ticker, so no possibility of it being called twice in a row.  What would happen if a user made multiple changes to the secret in quick succession?  Is there a possibility it could break something?